### PR TITLE
module/zstd: pass -U__BMI__

### PR DESCRIPTION
**Pull request by gyakovlev**
### Motivation and Context

fix for https://github.com/openzfs/zfs/issues/10758
if kernel is compiled with `-march=znver1` or `-march=znver2` zstd module compilation will fail due to SSE register return with SSE disabled.
what's interesting, is that `-march=skylake` also implies `-mbmi` which defines `__BMI__` but compilation succeeds.
It is probably due to different BMI implementations on AMD and INTEL processors and the way compiler uses instructions.

### Description

pass `-U__BMI__` while compiling zstd.

### How Has This Been Tested?

build-tested. error is gone. I can't actually run this as I don't have AMD ZEN CPU hardware.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
Would you mind adding this to the last line in module/Makefile.bsd as well? I added it in FreeBSD (we use a different Makefile for OpenZFS vendored in the base OS), but have not added it here yet.

**Comment by gyakovlev** [comment]
@freqlabs done!

File: module/zstd/Makefile.in [hunk 1]
```diff
@@ -20,6 +20,10 @@ ccflags-y += -O3
 # Set it for other compilers, too.
 $(obj)/lib/zstd.o: c_flags += -fno-tree-vectorize
 
+# SSE register return with SSE disabled if -march=znverX is passed
+# https://github.com/openzfs/zfs/issues/10758
```
Comment (behlendorf) [inline module/zstd/Makefile.in:24]: I don't think we need to explicitly reference the issue here.  The git commit comment will include the issue number for future reference.  Would you mind removing the link.

**Comment by BrainSlayer** [comment]
there is a indication that this is a compiler bug. the error is also not relevant for bmi instructions, its caused by a header defined function which is not related to bmi instructions. i dont know if this patch has any performance impact. but i will look if there is a better target or compiler version specific workaround

**Comment by freqlabs** [comment]
For reference, the purpose of undefining `__BMI__` (which is defined by the compiler when the target is set to something with support for BMI) is to nope out of this in module/zstd/lib/zstd.c:
```
/*=========================================
*  Target specific
=========================================*/
#if defined(__BMI__) && defined(__GNUC__)
#  include <immintrin.h>   /* support for bextr (experimental) */
```

**Comment by BrainSlayer** [comment]
@freqlabs i know. i just found out that this problem is a compiler glitch since it disables avx instructions for these cpus, so some inline functions within a specific unused header which is referenced by immintrin.h are compiled wrong.

**Pull request merged**